src/models/chan.js

```diff
@@ -183,21 +179,72 @@ Chan.prototype.getFilteredClone = function(lastActiveChannel, lastMessage) {
 	}, {});
 };
 
-function writeUserLog(client, msg) {
-	if (!msg.isLoggable()) {
-		return false;
+Chan.prototype.writeUserLog = function(client, msg) {
+	this.messages.push(msg);
+
+	// Does this user have logs disabled
+	if (!client.config.log) {
```

Comment:
I see you already do this check in `src/client.js` L87, should we do it again, or instead check that a message store was enabled?

Reply:
`src/client.js` check is for `sqlite` only, while this check here ensures no logging is performed for any method (sqlite or text). It will most likely become clearer when the logging API is abstracted out.